torch/_inductor/kernel/mm.py

```diff
@@ -339,6 +339,15 @@ def tuned_addmm(inp, mat1, mat2, *, alpha=1, beta=1, layout=None):
                 beta=beta,
             )
 
+    if use_ck_template(layout, m, n, k):
```

Comment:
So I've quickly scrolled through the `use_ck_template` and didn't notice a check that all input shapes are static. Should we add the check (and maybe others if relevant) here, similar to the CUTLASS backend above?

Reply:
I believe the check on `V.graph.sizevars.size_hint(m * n * k, fallback=-1)` falls back when the shape is dynamic?

Reply:
Yeah, I think it should be fine: https://github.com/pytorch/pytorch/blob/c4bf4005d1eafc755f0cdd2379ade0bc3bb3902c/torch/_inductor/sizevars.py#L484